plugin/src/main/java/dev/regadas/trino/pubsub/listener/SchemaHelpers.java

```diff
@@ -1,18 +1,37 @@
 package dev.regadas.trino.pubsub.listener;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import dev.regadas.trino.pubsub.listener.proto.Schema;
 import dev.regadas.trino.pubsub.listener.proto.Schema.TrinoWarning.Code;
 import io.trino.spi.WarningCode;
+import io.trino.spi.eventlistener.ColumnDetail;
+import io.trino.spi.eventlistener.OutputColumnMetadata;
 import io.trino.spi.eventlistener.QueryCompletedEvent;
 import io.trino.spi.eventlistener.QueryContext;
 import io.trino.spi.eventlistener.QueryCreatedEvent;
 import io.trino.spi.eventlistener.QueryFailureInfo;
+import io.trino.spi.eventlistener.QueryIOMetadata;
+import io.trino.spi.eventlistener.QueryInputMetadata;
 import io.trino.spi.eventlistener.QueryMetadata;
+import io.trino.spi.eventlistener.QueryOutputMetadata;
 import io.trino.spi.eventlistener.SplitCompletedEvent;
 import io.trino.spi.eventlistener.SplitFailureInfo;
+import java.util.List;
+import java.util.Optional;
 
 public final class SchemaHelpers {
 
+    private static ObjectMapper mapper = new ObjectMapper();
+
+    static Optional<String> jsonify(java.lang.Object obj) {
+        try {
+            return Optional.of(mapper.writeValueAsString(obj));
+        } catch (JsonProcessingException exc) {
+            return Optional.empty();
```

Comment:
in your view we don't care if this fails?

maybe we can bubble this up and fail to publish the message. Failed messages are logged

Reply:
There's two attributes that use it, `connectorInfo` and `connectorMetrics`.  Since they're expected to be provided over SPI, it can't be guaranteed that plugin implementations always add jackson annotations for their fields (the ones from Trino itself, [EG hive do though](https://github.com/trinodb/trino/blob/master/plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveInputInfo.java#L33-L35).)

Bailing the message entirely would mean a connector plugin that omits those annotations would fail fast, which could be a better behavior... I suppose the user of the event-listener plugin might not also be the author of the offending connector plugin though? Could lead to strictly incompatible setups.

Maybe we should log the exception with explanation above, but still return empty?

Reply:
Right, that's true plugins might not have these Jackson annotations, though Jackson should be able to serialize the object even if no annotations are present. I'm fine with at least adding a logger here to raise awareness about the missing fields.

Reply:
It is so! 😄